.github/workflows/check-nix-format.yml

```diff
@@ -0,0 +1,50 @@
+# This file was copied mostly from check-maintainers-sorted.yaml.
+# NOTE: Formatting with the RFC-style nixfmt command is not yet stable. See
+# https://github.com/NixOS/rfcs/pull/166.
+# Because of this, this action is not yet enabled for all files -- only for
+# those who have opted in.
+name: Check that Nix files are formatted
+
+on:
+  pull_request_target:
+permissions:
+  contents: read
+
+jobs:
+  nixos:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'NixOS'
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          # pull_request_target checks out the base branch by default
+          ref: refs/pull/${{ github.event.pull_request.number }}/merge
+      - uses: cachix/install-nix-action@8887e596b4ee1134dae06b98d573bd674693f47c # v26
+        with:
+          # explicitly enable sandbox
+          extra_nix_config: sandbox = true
+      - name: Install nixfmt
+        run: nix-env -f default.nix -iAP nixfmt-rfc-style
+      - name: Check that Nix files are formatted according to the RFC style
+        # Each environment variable beginning with NIX_FMT_PATHS_ is a list of
+        # paths to check with nixfmt.
+        env:
+          # Format paths related to the Nixpkgs CUDA ecosystem.
+          NIX_FMT_PATHS_CUDA: |
+            pkgs/development/cuda-modules
+            pkgs/test/cuda
+            pkgs/top-level/cuda-packages.nix
+        # Iterate over all environment variables beginning with NIX_FMT_PATHS_.
+        run: |
+          for env_var in "${!NIX_FMT_PATHS_@}"; do
```

Comment:
@ConnorBaker: Sorry if I am mistaken, but looking at this, I was think the following:
Why do we not make nix-shell or `nix-build` out of this, which has the following
- Format the files changed given a Git input to `git diff`, (bruteforce running on the repo)
- On `--check` given report if changes happened. Should run isolated in `/tmp` or so, because it does not alter files...

Is that the idea to implement this in `nixfmt`?

Reply:
Basically we only run `nix run nixfmt#check main...PR-123` or something

Reply:
That would be great functionality to have (I advocated for a GitHub action which does exactly that, maintained by the community), but I think it's outside the scope of this PR -- I want the files I'm working with most waking moments of my life to be formatted.

To be clear, I do think we should have a utility which allows people to do that, but that gets much closer to what `pre-commit` and friends do. I don't want this PR to turn into an RFC proposing such a workflow, which is why I'm content with a simple bash `for` loop.

Does that make sense @gabyx?

Reply:
Jeah sure, this `diffing` thing is only a CI issue, I would not go into `pre-commits`, they serve another matter, despite beeeing overlapping.

Since this is nixpkgs, the best would be at the end to have with the efforts @infinisil did, a tool which people can just run inside nixpkgs. e.g. `nix run .#format main...mybranch`. And as a bonus that will straight run inside this nixpkgs :).

@ConnorBaker : I am totally with you, formatting and having CI/Tooling working for such a big OSS project should be key and nobody wants to fight with stupidities as whitespace changes... (-> blame the tool, not the people) =)

Reply:
The changes you did to the Github workflow, I would generally consider good first solution. We could improve that to make it run directly with `nix` on and `alpine-with-nix` container or such... 
In the end we could add more and more paths to this list, and finally remove it and format the whole repo, when it works. @infinisil might have much more infos on that state etc, and maybe thats already the plan but way differently ;)